source/agora/consensus/state/ValidatorSet.d

```diff
@@ -237,6 +243,34 @@ public class ValidatorSet
         }
     }
 
+    /***************************************************************************
+
+        Check with public key if a enrollment data exists in the validator set
+
+        Params:
+            pubkey = the key by which the validator set searches enrollment
+
+        Returns:
+            true if the validator set has an enrollment for the public key
+
+    ***************************************************************************/
+
+    public bool hasPublicKey (in PublicKey pubkey) @trusted nothrow
+    {
+        try
+        {
+            auto results = this.db.execute("SELECT EXISTS(SELECT 1 FROM " ~
+                "validator_set WHERE public_key = ?)", pubkey.toString());
+            return results.front().peek!bool(0);
+        }
+        catch (Exception ex)
+        {
+            log.error("Exception occured on hasEnrollment: {}, " ~
```

Comment:
We need to stop writing code like this.

If there is an error in the database, then the enrollment will be accepted. Do you really want that to happen?

Reply:
> If there is an error in the database, then the enrollment will be accepted. Do you really want that to happen?

I added `assert(0)` statements into the two critical functions, `hasEnrollment` and `hasPublicKey`. But we would need to check more about other functions as a separate issue.

Reply:
Thanks. There is still an open issue about handling SQLite errors properly. Hopefully someone takes care of that.